torch/fx/graph.py

```diff
@@ -451,24 +451,28 @@ def append_stacktrace_summary(node : Node):
                         prev_stacktrace = node.stack_trace
 
                         lines = node.stack_trace.strip().split('\n')
-                        idx = 0
-                        while idx < len(lines):
+                        # stacktrace should have innermost frame last, so we
+                        # iterate backwards to find the first line that starts
+                        # with 'File '
+                        idx = len(lines) - 1
+                        while idx >= 0:
                             line = lines[idx].strip()
                             if line.startswith('File '):
                                 break
-                            idx += 1
+                            idx -= 1
```

Comment:
nit: You can use an else branch on the loop to elegantly set idx to None when it is not found

```
while ...:
    ...
else:
   idx = None
```